### PR TITLE
Delete and add version modal

### PR DESCRIPTION
**Pull request by chopachop**
## What is the purpose of this pull request?
Add modals for deletion and adding new versions

## What has changed?
- Added both modals
- Added new interface for post version request
- Added new ugeGetPageParams hook to get projectId from URL as a number
- adjustments and alignments in the hooks and components to make it all work

## Jira links
```jira_links
https://uniwise.atlassian.net/browse/MP-2462
https://uniwise.atlassian.net/browse/MP-2441
```

### CODE REVIEW

File: dashboard/src/api/hooks/useDeleteVersion.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { GetVersionsResponse } from '../../interfaces/versions';
+import client from '../client';
+import '../mocks/useDeleteVersion.mock';
+
+const deleteVersion = async (projectId?: number, versionId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useDeleteVersion.ts:6]: The props are optional. there is no need to call the server is we have undefined. You might be checking if projectid or versionid is not undefined before calling this function, but then you should remove the ?.

File: dashboard/src/api/hooks/useDeleteVersion.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const response = await client.delete<GetVersionsResponse>(
+    `/api/v1/projects/${projectId}/versions/${versionId}`,
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useDeleteVersion.ts:8]: there are a lot of places where we call /api/v1/projects -> create this as a constant instead.

File: dashboard/src/api/hooks/useDeleteVersion.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  );
+
+  return response.data;
+};
+
+export const useDeleteVersion = (projectId?: number, versionId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useDeleteVersion.ts:14]: same here with the ?

File: dashboard/src/api/hooks/useGetProject.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+import { Project } from '../../interfaces/projects';
+import client from '../client';
+import '../mocks/useGetProject.mock';
+
+const getProject = async (projectId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useGetProject.ts:6]: we should not set the optional

File: dashboard/src/api/hooks/useGetProject.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  const response = await client.get<Project>(
+    `/api/v1/projects/${projectId}`,
+  );
+
+  return response.data;
+};
+
+export const useGetProject = (projectId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useGetProject.ts:14]: we should not set the optional

File: dashboard/src/api/hooks/useGetVersions.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { useQuery } from '@tanstack/react-query';
+import { GetVersionsResponse } from '../../interfaces/versions';
+import client from '../client';
+import '../mocks/useGetVersions.mock';
+
+const getVersions = async (projectId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useGetVersions.ts:6]: we should not set the optional

File: dashboard/src/api/hooks/useGetVersions.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  const response = await client.get<GetVersionsResponse>(
+    `/api/v1/projects/${projectId}/versions`,
+  );
+
+  return response.data;
+};
+
+export const useGetVersions = (projectId?: number) => {
```
Comment (rimonnassory) [inline dashboard/src/api/hooks/useGetVersions.ts:14]: we should not set the optional

File: dashboard/src/components/TablePaginationSection.tsx [hunk 8]
```diff
@@ -2,8 +2,41 @@ import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
 import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
 import { Box, Button, IconButton, iconButtonClasses } from "@mui/joy";
 
-export const PaginationSection = () => {
-  //TODO: Implement pagination logic. Try to use cached data from query hook using query key
+interface TablePaginationSectionProps {
```
Comment (rimonnassory) [inline dashboard/src/components/TablePaginationSection.tsx:5]: i would have used type instead

**Review by rimonnassory — CHANGES_REQUESTED** [review]
There are a few things: 
1. create a constant for '/api/v1/projects/'
2. since you can do whatever you like with this project, i would recommend to use the library 'rect-hook-form' for your form. try it out!
3. a lot of places we are setting the parameters as ? when doing a request to the server. we should avoid that.

File: dashboard/src/api/hooks/useDeleteVersion.ts (hunk 1 above)
Comment (chopachop) [reply dashboard/src/api/hooks/useDeleteVersion.ts:6]: yeah totally makes sense

File: dashboard/src/api/hooks/useDeleteVersion.ts (hunk 2 above)
Comment (chopachop) [reply dashboard/src/api/hooks/useDeleteVersion.ts:8]: you mean something like const projects = '/api/v1/projects'?

File: dashboard/src/views/Projects/Versions/index.tsx [hunk 9]
```diff
@@ -1,49 +1,66 @@
...
+    const pageCount = Math.ceil(versionsList.versions.length / ITEMS_PER_PAGE);
+    const paginatedVersions = versionsList.versions.slice(
+      (currentPage - 1) * ITEMS_PER_PAGE,
+      currentPage * ITEMS_PER_PAGE
+    );
+    setPageCount(pageCount);
+    setPaginatedVersions(paginatedVersions);
+  }, [versionsList, currentPage]);
+
+  const handlePageChange = (newPage: number) => {
```
[the remark this answers is not in the document]
Comment (chopachop) [reply dashboard/src/views/Projects/Versions/index.tsx:61]: Yeah at some point a while ago we decided to put such onClick functionality into functions to have it all in one style

File: dashboard/src/components/TablePaginationSection.tsx [hunk 10]
```diff
@@ -2,8 +2,41 @@ import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
...
+  pageCount: number;
+  onPageChange: (page: number) => void;
+}
+
+export const TablePaginationSection: React.FC<TablePaginationSectionProps> = ({
+  currentPage,
+  pageCount,
+  onPageChange,
+}) => {
+  const generatePageNumbers = () => {
```
[the remark this answers is not in the document]
Comment (chopachop) [reply dashboard/src/components/TablePaginationSection.tsx:16]: Good idea! Will do

File: dashboard/src/components/TablePaginationSection.tsx [hunk 11]
```diff
@@ -2,8 +2,41 @@ import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
...
+    const pageNumbers = [];
+    if (pageCount <= 7) {
```
[the remark this answers is not in the document]
Comment (chopachop) [reply dashboard/src/components/TablePaginationSection.tsx]: tried to think about names for the constants and decided to go with explanatory comment at the top

File: dashboard/src/api/hooks/useGetProject.ts (hunk 4 above)
Comment (chopachop) [reply dashboard/src/api/hooks/useGetProject.ts:6]: we are having these to not have crutches inside component like another state that'll contain contain data and useEffect on top of that

**Review by shimaa-mohamed — APPROVED** [review]

**Review by naueramant — APPROVED** [review]

**Pull request merged**